widgetssdk/src/main/java/com/glia/widgets/entrywidget/EntryWidgetController.kt

```diff
@@ -1,109 +1,149 @@
 package com.glia.widgets.entrywidget
 
 import android.app.Activity
-import androidx.annotation.VisibleForTesting
 import com.glia.androidsdk.Engagement
 import com.glia.androidsdk.queuing.Queue
-import com.glia.androidsdk.queuing.QueueState
 import com.glia.widgets.chat.domain.IsAuthenticatedUseCase
 import com.glia.widgets.core.queue.QueueRepository
 import com.glia.widgets.core.queue.QueuesState
+import com.glia.widgets.core.secureconversations.domain.HasOngoingSecureConversationUseCase
 import com.glia.widgets.core.secureconversations.domain.ObserveUnreadMessagesCountUseCase
 import com.glia.widgets.di.GliaCore
 import com.glia.widgets.helper.Logger
 import com.glia.widgets.helper.TAG
+import com.glia.widgets.helper.mediaTypes
 import com.glia.widgets.launcher.EngagementLauncher
 import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
 import io.reactivex.rxjava3.core.Observable
 import io.reactivex.rxjava3.disposables.CompositeDisposable
 
-internal class EntryWidgetController(
+internal class EntryWidgetController @JvmOverloads constructor(
     private val queueRepository: QueueRepository,
     private val isAuthenticatedUseCase: IsAuthenticatedUseCase,
     private val observeUnreadMessagesCountUseCase: ObserveUnreadMessagesCountUseCase,
+    private val hasOngoingSecureConversationUseCase: HasOngoingSecureConversationUseCase,
     private val core: GliaCore,
-    private val engagementLauncher: EngagementLauncher
+    private val engagementLauncher: EngagementLauncher,
+    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
 ) : EntryWidgetContract.Controller {
+    private val queueStateObservable by lazy { queueRepository.queuesState.toObservable() }
+    private val unreadMessagesCountObservable by lazy { observeUnreadMessagesCountUseCase() }
+    private val hasOngoingSCObservable by lazy { hasOngoingSecureConversationUseCase() }
+
+    private val loadingState: List<EntryWidgetContract.ItemType> by lazy {
+        listOf(
+            EntryWidgetContract.ItemType.LoadingState,
+            EntryWidgetContract.ItemType.LoadingState,
+            EntryWidgetContract.ItemType.LoadingState,
+            EntryWidgetContract.ItemType.LoadingState,
+            EntryWidgetContract.ItemType.ProvidedBy
+        )
+    }
+
+    private val emptyState: List<EntryWidgetContract.ItemType.EmptyState> by lazy {
+        listOf(EntryWidgetContract.ItemType.EmptyState)
+    }
+
+    private val errorState: List<EntryWidgetContract.ItemType.EmptyState> by lazy {
+        listOf(EntryWidgetContract.ItemType.EmptyState)
+    }
+
+    private val messagingChatObservableItemType: Observable<List<EntryWidgetContract.ItemType>>
+        get() = queueStateObservable.map(::mapMessagingQueueState)
+
+    /** Actually the Bottom Sheet must be skipped when there is ongoing SC, so technically no need for ongoing SC check in this case.
```

Comment:
This sentence sounds contradictory to me because logically
ongoing SC check should be executed to know that "there is ongoing SC". That's why it sounds confusing to me that Bottom Sheet could be skipped without a check.

Reply:
Hm, maybe I misunderstood the comment. 🤔

Reply:
I will try to improve this comment, or maybe remove it because it was written particularly for the bottom sheet, but now the bottom sheet and embedded view share business logic.

Reply:
I decided to remove it. It is not relevant anymore.